### PR TITLE
Reset default value and generate unique name on parameter type change

### PR DESCRIPTION
**Pull request by uladzimirdev**
closes https://github.com/metabase/metabase/issues/41475

Reset default value and generate unique name for parameters with updated type.

broken e2e are expected to appear because of name change

https://github.com/metabase/metabase/assets/125459446/56020778-3e0f-45ef-99cf-44ccd74657a2

### CODE REVIEW

File: frontend/src/metabase/parameters/components/ParameterSettings/ParameterSettings.tsx [hunk 1]
```diff
@@ -123,10 +119,10 @@ export const ParameterSettings = ({
   const isMultiValue = getIsMultiSelect(parameter) ? "multi" : "single";
 
   const handleTypeChange = (sectionId: string) => {
-    // can be moved to redux?
-    const defaultOptionOfNextType = defaultOptionForSection[sectionId];
+    const option = defaultOptionForSection[sectionId];
 
-    onChangeType(defaultOptionOfNextType.type, sectionId);
+    onChangeType(option.type, sectionId);
+    onChangeName(generateParameterUniqueName(option));
```
[the remark this answers is not in the document]
Comment (uladzimirdev) [reply frontend/src/metabase/parameters/components/ParameterSettings/ParameterSettings.tsx:125]: good point

[the remark this answers is not in the document]
Comment (uladzimirdev) [reply frontend/src/metabase/parameters/components/ParameterSettings/ParameterSettings.tsx:125]: actually I see why I did like this - we also have a method called `handleOperatorChange` which uses inside `onChangeType(operatorType, sectionId as string);` and has access only to the type value and not whole the option, which is required for generating unique name. 

we don't have an optimal way to get option which corresponds to the operator type, so we'll need to go through the nested object and iterate over options every time, I think it requires more changes and more error prone approach

[the remark this answers is not in the document]
Comment (uladzimirdev) [reply frontend/src/metabase/parameters/components/ParameterSettings/ParameterSettings.tsx:125]: I actually agree with you here, previous solution with keeping the old name wasn't any better, so I asked product team if it's okay to change name. I'll double check with products

File: frontend/src/metabase/parameters/components/ParameterSettings/ParameterSettings.tsx [hunk 2]
```diff
@@ -123,15 +117,19 @@ export const ParameterSettings = ({
...
+    const option = defaultOptionForSection[sectionId];
 
-    onChangeType(defaultOptionOfNextType.type, sectionId);
+    onChangeType(option);
   };
 
   const handleOperatorChange = (operatorType: string) => {
-    // TODO: fix sectionId type
-    onChangeType(operatorType, sectionId as string);
+    const option = parameterSections
```
Comment (uladzimirdev) [inline frontend/src/metabase/parameters/components/ParameterSettings/ParameterSettings.tsx:126]: not optimal, so I created a followup https://github.com/metabase/metabase/issues/41517

File: frontend/src/metabase/parameters/components/ParameterSettings/ParameterSettings.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (uladzimirdev) [reply frontend/src/metabase/parameters/components/ParameterSettings/ParameterSettings.tsx]: @romeovs I reworked the approach and move logic re generating unique name to redux

[the remark this answers is not in the document]
Comment (uladzimirdev) [reply frontend/src/metabase/parameters/components/ParameterSettings/ParameterSettings.tsx]: @kamilmielnik I created a follow up task to preserve custom user name, I spent some time in scope of this task, but no luck to deliver a working solution

File: frontend/src/metabase/parameters/components/ParameterSettings/ParameterSettings.tsx [hunk 3]
```diff
@@ -123,15 +117,19 @@ export const ParameterSettings = ({
...
+      .find(section => section.id === sectionId)
+      ?.options.find(opt => opt.type === operatorType);
```
Comment (kamilmielnik) [inline frontend/src/metabase/parameters/components/ParameterSettings/ParameterSettings.tsx:128]: :eyes:

**Review by kamilmielnik — APPROVED** [review]

**Pull request merged**